src/gui/MainWindow.cpp

```diff
@@ -1729,10 +1730,15 @@ void MainWindow::hideYubiKeyPopup()
 void MainWindow::bringToFront()
 {
     ensurePolished();
-    setWindowState((windowState() & ~Qt::WindowMinimized) | Qt::WindowActive);
-    show();
-    raise();
-    activateWindow();
+    if (m_startMinimized) {
+        setWindowState(Qt::WindowMinimized);
+        showMinimized();
```

Comment:
This is very bad practice. This function does two opposite actions based on the state of a private variable. This needs to be in its own function or in a more appropriate one.